### DISCUSSION TITLE
Add new GalleryItemPropertySet to ItemsSource In Win10, looks incorrect.

### DISCUSSION THREAD

**Discussion by rulai-yinqiang** [harborsiem/WindowsRibbon#33] [Q&A]
`    Private Sub frmRibbon_Load(sender As Object, e As EventArgs) Handles Me.Load
        _dropDownGallery = New RibbonDropDownGallery(Ribbon1, 2)
        AddHandler _dropDownGallery.ItemsSourceReady, AddressOf _dropDownGallery_ItemsSourceReady
    End Sub

    Private Sub _dropDownGallery_ItemsSourceReady()
        Dim itemsSource As IUICollection = _dropDownGallery.ItemsSource

        itemsSource.Add(New GalleryItemPropertySet() With {.Label = "new item 1"})
        itemsSource.Add(New GalleryItemPropertySet() With {.Label = "new item 2"})
    End Sub`

![image](https://github.com/harborsiem/WindowsRibbon/assets/33405365/5137cef5-a469-43c5-8b50-18f23877ca3e)

BTW: It works good in Win11.

**Comment by harborsiem** [comment]
Please send the RibbonMarkup file and a more complete source code.

**Comment by rulai-yinqiang** [comment]
A full VB project is attached. I tested in Windows 10 19045.4529 Chinese version.

[WindowsApp10.zip](https://github.com/user-attachments/files/15981453/WindowsApp10.zip)

**Comment by harborsiem** [comment]
This case is crazy.
The RibbonDropDownGallery looks like TextPosition=Overlap in the markup file. But that is not the case.
So I checked the processor mode for the application. The processor mode was x86 at runtime (I checked the IntPtr Size property).
Then I changed AnyCpu to x64 and everything is looking fine. 
The same problem I can see also in a RibbonSplitButtonGallery.

Please could you check the processor mode when you run this application in Win11, because i have no Win11 installation.

**Comment by harborsiem** [comment]
I have done some tests for x86 code.
When I add a transparent image to the GalleryItemPropertySet's you defined in the example, then the user interface is looking correctly for the Label.

I think the issue you have found is a problem in the Microsoft Windows Ribbon Framework code for x86 processors.

In the zip file are 2 transparent .png image files which you can use with converting to IUIImage by the Ribbon method ConvertToUIImage(Bitmap).
[Transparent.zip](https://github.com/user-attachments/files/16010586/Transparent.zip)

**Comment by rulai-yinqiang** [comment]
I'm afraid add a Transparent bitmap is not a perfect solution.

In Windows 11,

Before add the Transparent bitmap, when I select a item, it may add a 'selected checkbox', looks great.
![image](https://github.com/harborsiem/WindowsRibbon/assets/33405365/d5e63cf1-bcf3-402e-8dee-b0bc77f559ff)

But after add a Transparent bitmap, when I select a item, it looks like this.
![image](https://github.com/harborsiem/WindowsRibbon/assets/33405365/31531ce6-a83f-4739-975b-240ff0c7b209)

BTW: 

In Windows 10, RibbonToggleButton (without image) in ItemsSource, when Toggled (BooleanValue = True), there's no 'selected checkbox' image.

But in Windows 11, it has a 'selected checkbox' image.

Looks same above.

**Reply by harborsiem** [reply]
> BTW:
> 
> In Windows 10, RibbonToggleButton (without image) in ItemsSource, when Toggled (BooleanValue = True), there's no 'selected checkbox' image.
> 
> But in Windows 11, it has a 'selected checkbox' image.

In this case the ItemsSource is set to GalleryType "Commands" in the markup?
I think Microsoft did an update in Windows 11 to the Ribbon Framework. But you can show a selected checkbox in Windows 10 by extra code with the help of ControlPaint.DrawCheckBox and setting SmallImage of RibbonToggleButton.

**Comment by harborsiem** [comment]
1. Did you use the *.png file in the constructor of the Bitmap class that I have send in the Transparent.zip file?
2. Or did you create a Bitmap with MakeTransparent?

The second one is not really a transparent Bitmap.

**Reply by rulai-yinqiang** [reply]
> 1. Did you use the *.png file in the constructor of the Bitmap class that I have send in the Transparent.zip file?
> 2. Or did you create a Bitmap with MakeTransparent?
> 
> The second one is not really a transparent Bitmap.

Yes, I used the Transparent_16.png file. I think the windows 11 ribbon framework will draw the 'checkbox' only if there's no image was set.

![image](https://github.com/harborsiem/WindowsRibbon/assets/33405365/8c4ebd89-7384-4572-b917-dae2e38ebcdc)

**Comment by harborsiem** [comment]
Transparency is lost when you copy to the resources.
You should create a Bitmap by code
```Vb
Dim image1 As Bitmap
image1 = New Bitmap("Transparent_16.png") ' Add a path !

... .ItemImage = Ribbon1.ConvertToUIImage(image1)})
```

**Comment by rulai-yinqiang** [comment]
Same in Windows 11, no check image after add a transparent image.
![image](https://github.com/harborsiem/WindowsRibbon/assets/33405365/f56fc779-2eca-4895-9bd3-632f9812d5af)

**Comment by harborsiem** [comment]
Ok, So we know there are some differences between Windows 10, Windows 11 and processor mode x64 and x86. For this issue I can't help anymore because the problem is in the Microsoft implementation.
If you find no workaround for this problem, then you can send a bug report to Microsoft.
